modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/SqlSchemaManagerImpl.java

```diff
@@ -79,11 +85,15 @@ public SqlSchemaManagerImpl(
         });
 
         schemasVv.whenComplete((token, stringIgniteSchemaMap, throwable) -> {
-            rebuild(token, stringIgniteSchemaMap);
+            if (throwable != null) {
+                throw new IgniteInternalException("Couldn't evaluate sql schemas for causality token: " + token, throwable);
```

Comment:
Why we need this? Who and how will handle such exception?

Reply:
moved this exception into `calciteSchemaVv.completeExceptionally`